src/parser.c

```diff
@@ -95,6 +95,54 @@ static int knot_value_as_int(const knot_value_type *data)
 	return data->val_i;
 }
 
+static char *get_str_value_from_json(json_object *jso, const char *key)
+{
+	char *str_value;
+	json_object *jobjkey;
+
+	if (!json_object_object_get_ex(jso, key, &jobjkey))
+		return NULL;
+
+	if (json_object_get_type(jobjkey) != json_type_string)
+		return NULL;
+	
+	str_value = l_strdup(json_object_get_string(jobjkey));
+	json_object_put(jso);
+	return str_value;
+}
+
+static void *device_array_item(json_object *array_item,
+		  	       create_device_item_cb cb)
+{
+	json_object *jobjkey;
+	struct l_queue *schema;
+	const char *id, *name;
+
+	/* Getting 'Id': Mandatory field for registered device */
+	id = get_str_value_from_json(array_item,
+		KNOT_JSON_FIELD_DEVICE_ID);
+	if (!id)
+		return NULL;
+
+	/* Getting 'schema': Mandatory field for registered device */
+	/* FIXME: Call a parser function instead of json-c function*/
+	if (!json_object_object_get_ex(array_item, 
```

Comment:
Please, remove this white space.
```suggestion
	if (!json_object_object_get_ex(array_item,
```